cmd/metrics-server/app/options/options.go

```diff
@@ -63,6 +63,27 @@ type Options struct {
 	DeprecatedCompletelyInsecureKubelet bool
 }
 
+func (o *Options) Validate() []error {
+	errors := []error{}
+	if o.KubeletCAFile != "" && o.InsecureKubeletTLS {
+		errors = append(errors, fmt.Errorf("Cannot use both --kubelet-ca-file and --kubelet-insecure-tls"))
+	}
+
+	if o.KubeletClientKeyFile == "" && o.KubeletClientCertFile == "" {
```

Comment:
```suggestion
	if (o.KubeletClientKeyFile == "") != (o.KubeletClientCertFile == "") {
```

Reply:
@serathius this a boolean type being comapres to a boolean type

Reply:
yes, left represent if KeyFile was set, right if CertFile was set. If they differ it means that either:
* --kubelet-client-key is set and --kubelet-client-certificate is not set
* reverse 

This is correct as our goal is to check if both flags are set at the same time